### ISSUE TITLE
Error 401 When using Python API

### ISSUE THREAD

**Issue by karuaan** [finnhubio/Finnhub-API#402]
Receiving Error Code 401 when trying to get Stocks Data from Finnhub. 

Currently, when I logged in to Finnhub, I can see the same API code and no invoice yet my account seems to be locked.

I am sure the API Key is working fine as it was working fine around 30 mins back. 

Account Email: karuaan@gmail.com

Error Code:

Traceback (most recent call last):
  File "/Users/karan/Desktop/Development_Projects/MyTradingProfile/Portfolio.py", line 177, in <module>
    currStockDetails = createStockDetails(curr_stocks)
  File "/Users/karan/Desktop/Development_Projects/MyTradingProfile/Portfolio.py", line 35, in createStockDetails
    stockPriceTargets = finnhub_client.price_target(stock)
  File "/opt/homebrew/lib/python3.9/site-packages/finnhub/client.py", line 124, in price_target
    return self._get("/stock/price-target", params={"symbol": symbol})
  File "/opt/homebrew/lib/python3.9/site-packages/finnhub/client.py", line 69, in _get
    return self._request("get", path, **kwargs)
  File "/opt/homebrew/lib/python3.9/site-packages/finnhub/client.py", line 39, in _request
    return self._handle_response(response)
  File "/opt/homebrew/lib/python3.9/site-packages/finnhub/client.py", line 44, in _handle_response
    raise FinnhubAPIException(response)
finnhub.exceptions.FinnhubAPIException: FinnhubAPIException(status_code: 401): You don't have access to this resource.

**Comment by finnhubio** [comment]
Hi there,

Thank you for posting an issue here. Which symbol are you trying to access ?

**Comment by karuaan** [comment]
Hi There,

I was going through a loop to check on the details for a set of tickers but I think was currently trying to access ABNB.

**Comment by karuaan** [comment]
ABNB is the first ticker in my list and the first check it should complete.

**Comment by finnhubio** [comment]
Hi,

Can you share the full request you are trying to make ( Endpoint and parameters etc.) ?

**Comment by karuaan** [comment]
# Setup Imports
finnhub_client = finnhub.Client(api_key=finnhub_api_key)
yec = yahoo_earnings_calendar.YahooEarningsCalendar()

curr_stocks = ['ABNB','DIS','UBER','SNOW','EXPE','BABA','BA','DDOG','LUV','TGT','ALK','CVNA']

watchlist_stocks = ['XOM','JNJ','NFLX','PINS','PYPL','NVDA','V']

def createStockDetails(stocks):
    
    stockStatistics = []

    for stock in stocks:

        print(stock)

        stockDetails = {}
        priceTargets = {}
        stockStats = {}
        currentPrice = {}
        stockRecommendations = {}
        stockAnalytics = {}

        # Finnhub Price Target Data

        stockPriceTargets = finnhub_client.price_target(stock)

        pricetargetshigh = stockPriceTargets.get('targetHigh')
        pricetargetslow = stockPriceTargets.get('targetLow')
        pricetargetsmean = stockPriceTargets.get('targetMean')
        pricetargetsmedian = stockPriceTargets.get('targetMedian')
        deviationInHighPrice = ((pricetargetshigh - pricetargetsmedian) / pricetargetsmedian) * 100
        deviationInLowPrice = ((pricetargetsmedian - pricetargetslow) / pricetargetsmedian) * 100

        priceTargets["HIGH"] = pricetargetshigh
        priceTargets["LOW"] = pricetargetslow
        priceTargets["MEAN"] = pricetargetsmean
        priceTargets["MEDIAN"] = pricetargetsmedian
        priceTargets["High Median Deviation"] = deviationInHighPrice
        priceTargets["High Low Deviation"] = deviationInLowPrice

        
        # Stock Prices for Today
        
        stockCurrentPrice = finnhub_client.quote(stock)

        closePrice = stockCurrentPrice.get('c')
        highPrice = stockCurrentPrice.get('h')
        lowPrice = stockCurrentPrice.get('l')
        openPrice = stockCurrentPrice.get('o')
        previousClose = stockCurrentPrice.get('pc')
        changeOpenToClose = ((closePrice - openPrice) / openPrice) * 100
        changeHighLow = ((highPrice - lowPrice) / lowPrice) * 100
        changeTodayYesterday = ((closePrice - previousClose) / 100) * 100

        currentPrice["Open"] = openPrice
        currentPrice["Close"] = closePrice
        currentPrice["High"] = highPrice
        currentPrice["Low"] = lowPrice
        currentPrice["Previous Close"] = previousClose
        currentPrice["Change Open to Close"] = changeOpenToClose
        currentPrice["Change High Low"] = changeHighLow
        currentPrice["Change Today Close Yesterday Close"] = changeTodayYesterday

        # Stock Information

        stockProfile = finnhub_client.company_profile2(symbol=stock)

        outstandingShares = stockProfile.get('shareOutstanding')
        marketCap = stockProfile.get('marketCapitalization')
        industry = stockProfile.get('finnhubIndustry')
        name = stockProfile.get('name')

         # Next Earning Date

        nextDateUnix = int(yec.get_next_earnings_date(stock))
        nextDateFormatted = datetime.datetime.utcfromtimestamp(nextDateUnix).strftime('%m-%d-%Y')

        stockStats['Name'] = name
        stockStats['Industry'] = industry
        stockStats['Market Cap'] = marketCap
        stockStats['Outstanding Shares'] = outstandingShares
        stockStats['Next Earnings Date'] = nextDateFormatted

        # Stock Recommendations

        stockRecs = finnhub_client.recommendation_trends(stock)

        strongBuy = stockRecs[0].get('strongBuy')
        buys = stockRecs[0].get('buy')
        holds = stockRecs[0].get('hold')
        sell = stockRecs[0].get('sell')
        strongSells = stockRecs[0].get('sell')

        stockRecommendations['Strong Buy'] = stockRecs[0].get('strongBuy')
        stockRecommendations['Buy'] = stockRecs[0].get('buy')
        stockRecommendations['Hold'] = stockRecs[0].get('hold')
        stockRecommendations['Sell'] = stockRecs[0].get('sell')
        stockRecommendations['Strong Sell'] = stockRecs[0].get('strongSell')
        stockRecommendations['Last Update'] = stockRecs[0].get('period')

        # Related Stocks

        related_stocks_list = finnhub_client.company_peers(stock)

        # Stock Analysis

        currPriceMedian = ((closePrice - pricetargetsmedian) / pricetargetsmedian)
        currPriceLow = ((closePrice - pricetargetslow) / pricetargetslow)
        currPriceHigh = ((closePrice - pricetargetshigh) / pricetargetshigh)
        totalRecs = strongBuy + buys + holds + sell + strongSells
        buyRating = (((strongBuy * 2) +  buys) / totalRecs) * 100
        holdRating = (holds / totalRecs) * 100
        sellRating = (((strongSells * 2) + sell) / totalRecs) * 100

        stockAnalytics["Price to High Target Deviation"] = currPriceHigh
        stockAnalytics["Price to Median Target Deviation"] = currPriceMedian
        stockAnalytics["Price to Low Target Deviation"] = currPriceLow
        stockAnalytics["Buy Rating"] = buyRating
        stockAnalytics["Hold Rating"] = holdRating
        stockAnalytics["Sell Rating"] = sellRating

        stockDetails['Ticker'] = stock
        stockDetails['Stock Information'] = stockStats
        stockDetails['Stock Pricing'] = currentPrice
        stockDetails['Stock Analysis'] = stockAnalytics
        stockDetails['Stock Price Targets'] = priceTargets
        stockDetails['Stock Recommendations'] = stockRecommendations
        stockDetails['Related Stocks'] = related_stocks_list

        stockStatistics.append(stockDetails)
    return stockStatistics

**Comment by karuaan** [comment]
It gets the key from another file.

**Comment by finnhubio** [comment]
I see. Thank you for the details. Following our email announcement this week, we are moving the Price target endpoint to the Premium Tier so you might not be able to access it with a free account. 

We, however, have decided to keep the Earnings calendar endpoint free after receiving feedback from the community since the announcement.

Let me know if you have any further questions.

**Comment by karuaan** [comment]
I see, let me try again.

**Comment by karuaan** [comment]
Worked. It was due to the price targets. Thank You for the help.

**Comment by finnhubio** [comment]
It's my pleasure. I'm glad to hear it's working for you now. Feel free to reach out again here or via email should you have any further questions.

**Comment by karuaan** [comment]
1 small question regarding the company profile 2 endpoint, what are the units for these (Like in thousands or millions): 

marketCapitalization
Market Capitalization. --> Is this number in thousands or million or etc.

shareOutstanding
Number of oustanding shares. --> Is this number in thousands or million or etc.

**Comment by finnhubio** [comment]
It's in millions

**Comment by karuaan** [comment]
Thank You. Closing the issue now thank you for the help.
